configuration/types.go

```diff
@@ -414,6 +414,21 @@ type Configuration struct {
 	// but can use 10s of GBs of RAM, even with pruning enabled.
 	MemoryLimitDisabled bool `json:"memory_limit_disabled"`
 
+	// AllInMemoryDisabled configures storage to increase memory
+	// usage. Enabling this massively increases performance
+	// but can use >20s of GBs of RAM, even with pruning enabled.
+	AllInMemoryEnabled bool `json:"all_in_memory_enabled"`
+
+	// TableSize unit is GB, enable users to define MaxTableSize
+	// when AllInMemoryEnabled == true or MemoryLimitDisabled== true, Cli will look up this config
+	// default value is 6, modification range is [2, 100]
+	TableSize *int64 `json:"table_size,omitempty"`
```

Comment:
Two questions about the data type:
1. If `TableSize` can't be negative, why we still use `int64` instead of `uint64`?
2. Why we use pointer here?